mesonbuild/options.py

```diff
@@ -774,23 +744,39 @@ def get(self, *args, **kwargs) -> UserOption:
 
     def is_project_option(self, key: OptionKey) -> bool:
         """Convenience method to check if this is a project option."""
-        return key.type is OptionType.PROJECT
+        return key in self.project_options
 
     def is_reserved_name(self, key: OptionKey) -> bool:
-        return not self.is_project_option(key)
+        if key.name in _BUILTIN_NAMES:
+            return True
+        # FIXME, this hack is needed until the lang field is removed from OptionKey.
+        if key.lang is not None:
+            return True
+        if '_' not in key.name:
+            return False
+        prefix = key.name.split('_')[0]
+        # Pylint seems to think that it is faster to build a set object
+        # and all related work just to test whether a string has one of two
+        # values. It is not, thank you very much.
+        if prefix in ('b', 'backend'): # pylint: disable=R6201
```

Comment:
pylint is correct. Because this set would be a compiled constant python will convert the raw set into a pre-computed frozen set at byte compile time.

Reply:
Even so, doing the lookup requires you to compute the hash code and do one or two comparisons. With a tuple it is just one or two string comparisons. Computing the hash code is a lot heavier operation than string comparisons.

The only way it could be more efficient would be if the implementation skips hashes when the size is small. The documentation does not seem to explicitly say that is what happens.